### PR TITLE
Use Clock class throughout stats implementation.

### PR DESCRIPTION
**Pull request by sebright**

### CODE REVIEW

File: core/src/main/java/com/google/instrumentation/internal/TestClock.java [hunk 1]
```diff
@@ -28,14 +28,15 @@
   private static final int NUM_NANOS_PER_SECOND = 1000 * 1000 * 1000;
 
   @GuardedBy("this")
-  private Timestamp currentTime = Timestamp.create(0, 0);
+  private Timestamp currentTime = Timestamp.create(123, 456);
```
Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/internal/TestClock.java]: Can we use a more realistic timestamp? Some timestamp in 2017?

File: core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    return INSTANCE;
+  }
+
+  @Override
+  public Timestamp now() {
+    //    // TODO(sebright): Use Instant once we skip building this class with Java 7.
+    //    Instant instant = Instant.now();
+    //    return Timestamp.create(instant.getEpochSecond(), instant.getNano());
+
+    return Timestamp.fromMillis(System.currentTimeMillis());
```
Comment (bogdandrutu) [inline core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java]: Why not using Instant.now()? Already changed to use Java 8.

File: core_impl/src/test/java/com/google/instrumentation/stats/MeasurementDescriptorToViewMapTest.java [hunk 3]
```diff
@@ -15,30 +15,31 @@
...
-            RpcMeasurementConstants.RPC_CLIENT_ROUNDTRIP_LATENCY.getMeasurementDescriptorName(),
-            expected);
-    View actual = measurementDescriptorToViewMap.getView(
-        RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW);
+        RpcMeasurementConstants.RPC_CLIENT_ROUNDTRIP_LATENCY.getMeasurementDescriptorName(),
+        expected);
+    View actual =
+        measurementDescriptorToViewMap.getView(
+            RpcViewConstants.RPC_CLIENT_ROUNDTRIP_LATENCY_VIEW, clock);
     assertThat(actual.getViewDescriptor()).isEqualTo(expected.getViewDescriptor());
```
Comment (songy23) [inline core_impl/src/test/java/com/google/instrumentation/stats/MeasurementDescriptorToViewMapTest.java]: Also verify the start and end timestamp between expected and actual view?

File: core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java [hunk 4]
```diff
@@ -54,7 +56,7 @@
 
 
   private final StatsManagerImplBase statsManager =
-      new StatsManagerImplBase(new SimpleEventQueue());
+      new StatsManagerImplBase(new SimpleEventQueue(), TestClock.create(Timestamp.create(1, 2)));
```
Comment (songy23) [inline core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java]: Could we store a reference of this TestClock and compare the actual start and end time of views against it for the tests below? (line 101)

File: core/src/main/java/com/google/instrumentation/internal/TestClock.java (hunk 1 above)
Comment (sebright) [inline core/src/main/java/com/google/instrumentation/internal/TestClock.java]: Done.  I was also wondering if we should remove the no-arg factory method and use a different clock subclass for tests that don't need to set the time.

File: core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java (hunk 2 above)
Comment (sebright) [inline core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java]: I would need to update the Travis build to not build core_impl_java_8 with Java 7.  I wanted to do that in a separate PR.

File: core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java (hunk 4 above)
Comment (sebright) [inline core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java]: Done

File: core_impl/src/test/java/com/google/instrumentation/stats/MeasurementDescriptorToViewMapTest.java (hunk 3 above)
Comment (sebright) [inline core_impl/src/test/java/com/google/instrumentation/stats/MeasurementDescriptorToViewMapTest.java]: Done

File: core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  @Override
+  public long nowNanos() {
+    // TODO(sebright): Is this correct?
+    return System.nanoTime();
```
Comment (sebright) [inline core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java]: @bogdandrutu Is this the implementation we should use in Java 8?

Comment (bogdandrutu) [inline core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java]: Yes.

File: core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java (hunk 2 above)
Comment (bogdandrutu) [inline core_impl_java_8/src/main/java/com/google/instrumentation/internal/InstantClock.java]: Sure.

**Comment by sebright** [comment]
Thanks!
